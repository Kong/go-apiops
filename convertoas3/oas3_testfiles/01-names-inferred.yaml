# inferring names from the document title, the path+method and/or operation id

openapi: '3.0.0'
info:
  # document name detected from the title below
  title: Simple API overview
  version: v2
servers:
  - url: http://backend1.com/some/path
  - url: http://backend2.com/some/path
paths:
  /:
    # path with only a single '/'
    get:
      # operation with an ID
      operationId: opsid1
      responses: 
        '200':
          description: 200 ok
    post:
      # operation without an ID
      responses: 
        '200':
          description: 200 ok
  /application:
    # non-empty path
    get:
      # operation with an ID
      operationId: opsid2
      responses: 
        '200':
          description: 200 ok
    post:
      # operation without an ID
      responses: 
        '200':
          description: 200 ok


