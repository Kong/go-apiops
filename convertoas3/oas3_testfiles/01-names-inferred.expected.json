{
  "_format_version": "3.0",
  "services": [
    {
      "host": "simple-api-overview.upstream",
      "id": "0907c4ab-d9e4-5d21-813b-c57a97eeaad9",
      "name": "simple-api-overview",
      "path": "/some/path",
      "plugins": [],
      "port": 80,
      "protocol": "http",
      "routes": [
        {
          "id": "6fb3ba5b-774a-5b28-aa3c-ab9c6a26b484",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_opsid1",
          "paths": [
            "~/$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_01-names-inferred.yaml"
          ]
        },
        {
          "id": "2ab3e49d-7565-5ac2-aaee-18d060e2e712",
          "methods": [
            "POST"
          ],
          "name": "simple-api-overview_~_post",
          "paths": [
            "~/$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_01-names-inferred.yaml"
          ]
        },
        {
          "id": "fc7203a1-3b29-5eac-ac56-a1d361e14d97",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_opsid2",
          "paths": [
            "~/application$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_01-names-inferred.yaml"
          ]
        },
        {
          "id": "f388efcc-933e-54d5-a549-2b27ef4b935f",
          "methods": [
            "POST"
          ],
          "name": "simple-api-overview_application_post",
          "paths": [
            "~/application$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_01-names-inferred.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_01-names-inferred.yaml"
      ]
    }
  ],
  "upstreams": [
    {
      "id": "811c42d6-ef18-5296-a550-7dca2262b4d8",
      "name": "simple-api-overview.upstream",
      "tags": [
        "OAS3_import",
        "OAS3file_01-names-inferred.yaml"
      ],
      "targets": [
        {
          "tags": [
            "OAS3_import",
            "OAS3file_01-names-inferred.yaml"
          ],
          "target": "backend1.com:80"
        },
        {
          "tags": [
            "OAS3_import",
            "OAS3file_01-names-inferred.yaml"
          ],
          "target": "backend2.com:80"
        }
      ]
    }
  ]
}
